src/index.mjs

```diff
@@ -35,85 +35,82 @@ const RETURN = 13;
 const SPACE = 32;
 const SLASH = 47;
 
-const MODE_WHITESPACE = 0;
-const MODE_TEXT = 1;
+const MODE_TEXT = 0;
+const MODE_WHITESPACE = 1;
 const MODE_TAGNAME = 9;
 const MODE_ATTRIBUTE = 13;
 const MODE_SKIP = 47;
 
 /** Create a template function given strings from a tagged template. */
 const build = (statics) => {
-	let mode = MODE_WHITESPACE;
+	let mode = MODE_TEXT;
 	let out = 'return ';
 	let buffer = '';
-	let field = '';
-	let hasChildren = 0;
+	let tagClose = '';
+	let childClose = '';
 	let props = '';
 	let propsClose = '';
 	let spreadClose = '';
 	let quote = 0;
-	let slash, charCode, inTag, propName, propHasValue;
+	let fallbackPropValue = true;
+	let charCode, propName;
 
-	const commit = () => {
-		if (!inTag) {
+	const commit = field => {
+		if (mode === MODE_TEXT) {
 			if (field || (buffer = buffer.replace(/^\s*\n\s*|\s*\n\s*$/g,''))) {
-				if (hasChildren++) out += ',';
-				out += field || stringify(buffer);
+				out += childClose + (field || stringify(buffer));
+				childClose = ',';
 			}
 		}
 		else if (mode === MODE_TAGNAME) {
-			if (hasChildren++) out += ',';
-			out += 'h(' + (field || stringify(buffer));
+			out += childClose + 'h(' + (field || stringify(buffer));
+			childClose = ',';
 			mode = MODE_WHITESPACE;
 		}
-		else if (mode === MODE_ATTRIBUTE || (mode === MODE_WHITESPACE && buffer === '...')) {
+		else if (mode === MODE_WHITESPACE && buffer === '...') {
+			if (!spreadClose) {
+				props = 'Object.assign(' + (props || '{}');
+			}
+			props += propsClose + ',' + (field || '');
+			propsClose = '';
+			spreadClose = ')';
+		}
+		else if (mode === MODE_ATTRIBUTE || mode === MODE_WHITESPACE) {
 			if (mode === MODE_WHITESPACE) {
-				if (!spreadClose) {
-					spreadClose = ')';
-					if (!props) props = 'Object.assign({}';
-					else props = 'Object.assign(' + props;
-				}
-				props += propsClose + ',' + field;
-				propsClose = '';
+				propName = buffer;
+				buffer = '';
 			}
-			else if (propName) {
-				if (!props) props += '{';
-				else props += ',' + (propsClose ? '' : '{');
+			
+			if (propName) {
+				if (!props) {
+					props += '{';
+				}
+				else {
+					props += ',' + (propsClose ? '' : '{');
+				}
 				propsClose = '}';
-				props += stringify(propName) + ':';
-				props += field || ((propHasValue || buffer) && stringify(buffer)) || 'true';
+				props += stringify(propName) + ':' + (field || stringify(buffer || fallbackPropValue));
 				propName = '';
 			}
-			propHasValue = false;
-		}
-		else if (mode === MODE_WHITESPACE) {
-			mode = MODE_ATTRIBUTE;
-			// we're in an attribute name
-			propName = buffer;
-			buffer = field = '';
-			commit();
-			mode = MODE_WHITESPACE;
+			fallbackPropValue = true;
 		}
-		buffer = field = '';
+		buffer = '';
 	};
 
 	for (let i=0; i<statics.length; i++) {
 		if (i > 0) {
```

Comment:
```suggestion
 		if (i) {
```

Reply:
Big fan of the size optimizations here. Just thought maybe we could save one more byte?

Reply:
Good catch 👍

**Before change:**
```
Build "htm" to dist:
        605 B: htm.mjs.gz
        546 B: htm.mjs.br
        670 B: htm.umd.js.gz
        599 B: htm.umd.js.br
```

**After change:**
```
Build "htm" to dist:
        603 B: htm.mjs.gz
        544 B: htm.mjs.br
        669 B: htm.umd.js.gz
        595 B: htm.umd.js.br
```

Reply:
Pushed this change (thanking you in the commit message, if that's ok :)